extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/spi/datafetcher/BlockingHelper.java

```diff
@@ -4,27 +4,37 @@
 
 import io.smallrye.graphql.schema.model.Execute;
 import io.smallrye.graphql.schema.model.Operation;
-import io.vertx.core.Context;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
 import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
 
 public final class BlockingHelper {
 
     private BlockingHelper() {
     }
 
-    public static boolean blockingShouldExecuteNonBlocking(Operation operation, Context vc) {
+    public static boolean blockingShouldExecuteNonBlocking(Operation operation) {
         // Rule is that by default this should execute blocking except if marked as non-blocking)
         return operation.getExecute().equals(Execute.NON_BLOCKING);
     }
 
-    public static boolean nonBlockingShouldExecuteBlocking(Operation operation, Context vc) {
+    public static boolean nonBlockingShouldExecuteBlocking(Operation operation) {
         // Rule is that by default this should execute non-blocking except if marked as blocking
-        return operation.getExecute().equals(Execute.BLOCKING) && vc.isEventLoopContext();
+        return operation.getExecute().equals(Execute.BLOCKING) && Vertx.currentContext().isEventLoopContext();
     }
 
     @SuppressWarnings("unchecked")
-    public static void runBlocking(Context vc, Callable<Object> contextualCallable, Promise result) {
+    public static void runBlocking(Callable<Object> contextualCallable, Promise result) {
         // Here call blocking with context
-        vc.executeBlocking(contextualCallable).onComplete(result);
+        Uni.createFrom().item(contextualCallable)
+                .runSubscriptionOn(Infrastructure.getDefaultExecutor())
```

Comment:
This will loose the duplicated context propagation.

Why do we have a case where the duplicated context is `null`? What's the entry point?

Reply:
It's when `CompletableFuture.supplyAsync` is used in a method that has a batched source field that is blocking.

Reply:
`supplyAsync` without executor?

Reply:
We need to be very careful here and probably try to capture the initial duplicated context and restore it once the field is retrieved. Otherwise, you will likely be unable to handle security restrictions, tracing, and so on.